contentcuration/contentcuration/views/base.py

```diff
@@ -140,21 +125,35 @@ def get_or_set_cached_constants(constant, serializer):
 @has_accepted_policies
 @permission_classes((AllowAny,))
 def channel_list(request):
-    anon = request.user.is_anonymous()
+    anon = settings.LIBRARY_MODE or request.user.is_anonymous()
     current_user = (
         None
         if anon
         else json_for_parse_from_serializer(UserChannelListSerializer(request.user))
     )
     preferences = DEFAULT_USER_PREFERENCES if anon else request.user.content_defaults
+
+    # Get public channel languages
+    public_lang_query = Language.objects.filter(channel_language__public=True,
+                                                channel_language__main_tree__published=True,
+                                                channel_language__deleted=False) \
+                                        .values('lang_code') \
+                                        .annotate(count=Count('lang_code')) \
+                                        .order_by('lang_code')
+
+    # Get public channel sets
+    public_channelset_query = ChannelSet.objects.filter(public=True).annotate(public_only=Value(True, BooleanField()))
```

Comment:
This is only one, so shouldn't be an issue right now, but the serializer will cause `N` queries because of the query for the token, could you open a follow up issue to handle this in the future?